patches/tModLoader/Terraria/Wiring.cs.patch

```diff
@@ -50,6 +50,27 @@
  		private static int CurrentUser = 254;
  
  		public static void SetCurrentUser(int plr = -1) {
+@@ -226,7 +_,8 @@
+ 
+ 		public static bool Actuate(int i, int j) {
+ 			Tile tile = Main.tile[i, j];
+-			if (!tile.actuator())
++
++			if (!tile.actuator() || !TileLoader.CanActuate(tile))
+ 				return false;
+ 
+ 			if ((tile.type != 226 || !((double)j > Main.worldSurface) || NPC.downedPlantBoss) && (!((double)j > Main.worldSurface) || NPC.downedGolemBoss || Main.tile[i, j - 1].type != 237)) {
+@@ -241,6 +_,10 @@
+ 
+ 		public static void ActuateForced(int i, int j) {
+ 			Tile tile = Main.tile[i, j];
++
++			if (!TileLoader.CanActuate(tile))
```

Comment:
Are you sure this should be the same hook, or should it be a new hook? `CanActuateForcibly`, you can have that by default call `CanActuate`, which gives a bit finer control.